src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs

```diff
@@ -78,6 +84,32 @@ private Task<Signature> CreateSignature(X509Certificate2 cert, SignatureManifest
             return Task.FromResult(Signature.Load(cms));
         }
 
+        private Task<Signature> CreateSignature(
+            X509Certificate2 cert,
+            SignatureManifest signatureManifest,
+            CngKey privateKey,
+            Common.HashAlgorithmName hashAlgorithm,
+            CryptographicAttributeObjectCollection attributes
+            )
+        {
+            var cms = NativeUtilities.NativeSign(
+                signatureManifest.GetBytes(), cert, privateKey, attributes, hashAlgorithm);
+
+            return Task.FromResult(Signature.Load(cms));
+        }
+
+        private CryptographicAttributeObjectCollection GetSignAttributes()
+        {
+            var attributes = new CryptographicAttributeObjectCollection();
+
+            var signingTime = new Pkcs9SigningTime();
+            attributes.Add(new CryptographicAttributeObject(
+                    signingTime.Oid,
+                    new AsnEncodedDataCollection(signingTime)));
```

Comment:
Lines 105-108 can be simplified to:

    attributes.Add(new Pkcs9SigningTime());